openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -394,7 +394,25 @@ def _on_user_authentication_failed(request):
         if is_request_from_mobile_app(request):
             return HttpResponse(status=401)
 
-        return redirect_to_login(request.path)
+        # .. toggle_name: REDIRECT_TO_LOGIN_ON_SAFE_SESSION_AUTH_FAILURE
+        # .. toggle_implementation: SettingToggle
+        # .. toggle_default: False
+        # .. toggle_description: Turn this toggle on if you want to redirect_to_login on every request
+        #      in case of auth faliure in safe session middlewware.
+        # .. toggle_use_cases: temporary
+        # .. toggle_creation_date: 2021-10-18
+        # .. toggle_target_removal_date: 2021-10-22
+        # .. toggle_tickets: https://openedx.atlassian.net/browse/ARCHBOM-1911
+        REDIRECT_TO_LOGIN_ON_SAFE_SESSION_AUTH_FAILURE = getattr(settings,
+                                                                 'REDIRECT_TO_LOGIN_ON_SAFE_SESSION_AUTH_FAILURE',
+                                                                 False
+                                                                 )
+
+        if REDIRECT_TO_LOGIN_ON_SAFE_SESSION_AUTH_FAILURE or 'text/html' in request.MEATA.get('HTTP_ACCEPT', ''):
+            set_custom_attribute("safe_sessions.auth_failure", "redirect_to_login")
+            return redirect_to_login(request.path)
+        set_custom_attribute("safe_sessions.auth_failure", "401")
+        return HttpResponse(status=401)
```

Comment:
Notes to self for later (I need to stop now):
1. How does HttpResponse work for different accept headers? When would we want expect a JSON response?
2. Could this replace the need for `is_request_from_mobile_app` above?

Reply:
From a security perspective (maybe?) we try to minimize details shared about authN and authZ errors. So, maybe this is a bad idea. But from an Observability perspective, I sure wish we had an easier way to find out where a 401 or 403 was coming from. This is giving me some ideas.

As far as this PR is concerned, you can ignore this.